instaloader/structures.py

```diff
@@ -132,7 +132,10 @@ def __hash__(self) -> int:
     def _obtain_metadata(self):
         if not self._full_metadata_dict:
             pic_json = self._context.get_json("p/{0}/".format(self.shortcode), params={})
-            self._full_metadata_dict = pic_json['entry_data']['PostPage'][0]['graphql']['shortcode_media']
+            try:
+                self._full_metadata_dict = pic_json['entry_data']['PostPage'][0]['graphql']['shortcode_media']
```

Comment:
Like this
````python
if 'graphql' in pic_json:
    self._full_metadata_dict = pic_json['graphql']['shortcode_media']
else:
    self._full_metadata_dict = pic_json['entry_data']['PostPage'][0]['graphql']['shortcode_media']

````

Reply:
Thanks It make way more sense like that 👍